gossip/client_test.go

```diff
@@ -138,26 +138,32 @@ func TestClientGossip(t *testing.T) {
 		}
 	}()
 
-	if err := local.AddInfo("local-key", nil, time.Second); err != nil {
+	if err := local.AddInfo("local-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
-	if err := remote.AddInfo("remote-key", nil, time.Second); err != nil {
+	if err := remote.AddInfo("remote-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
 
 	// Use an insecure context. We're talking to tcp socket which are not in the certs.
 	rpcContext := rpc.NewContext(&base.Context{Insecure: true}, nil, stopper)
-	client.start(local, disconnected, rpcContext, stopper)
+	disconnected <- client
 
-	util.SucceedsSoon(t, func() error {
```

Comment:
> IMO, using SucceedsSoon like this is an anti-pattern

What is "like this"? By this logic, when would `SucceedsSoon` be appropriate? I'd rather increase the timeout on `SucceedsSoon` than move away from it; it's a much nicer experience than letting the test runner panic.